### PR TITLE
refactor(app): Standardize page and modal components

### PR DESCRIPTION
**Pull request by Kadee80**
## overview

This PR is an initial pass at standardizing pages, title bars, and modals in the app. Addresses weird scroll behavior with modals and pages on smaller screens.
## changelog

- Page component now receives titleBar and modal props along with children.
- Alert modals center on page, modal page modals align top. 

## review requests

Please click though all screens on app and make everything looks as expected. Try resizing and scrolling modals and page content.

You will notice ModalPages' content now aligns on the top rather than centering. This is consistent with designs in zeplin.

This was a weird refactor so please keep an eye out of redundant CSS.

### CODE REVIEW

File: app/src/components/Page.css [hunk 1]
```diff
@@ -1,5 +1,9 @@
 .task {
-  flex: 1 1;
+  flex: 1 1 calc(100% - 22rem);
```
Comment (mcous) [inline app/src/components/Page.css]: If this 22rem is coming from the nav bar and side panel, is there any way we can get those widths in vars?

File: app/src/components/Page.js [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
 
-import styles from './Page.css'
+import {TitleBar, type TitleBarProps} from '@opentrons/components'
 import LostConnectionAlert from './LostConnectionAlert'
 import {AnalyticsSettingsModal} from './analytics-settings'
+import styles from './Page.css'
+
+type Props = {
+  titleBar?: TitleBarProps,
+  children: any,
```
Comment (mcous) [inline app/src/components/Page.js]: Should be `React.Node`

File: app/src/components/Page.js [hunk 3]
```diff
@@ -1,14 +1,26 @@
...
 
-export default function Page (props) {
+export default function Page (props: Props) {
+  const {titleBar, children, modals} = props
   return (
     <main className={styles.task}>
-      {props.children}
+      {titleBar && (<TitleBar {...titleBar} />)}
+      <div className={styles.main}>
+      {children}
```
Comment (mcous) [inline app/src/components/Page.js]: Only because I'm already requesting changes in this file: can you fix up the indentation here?

File: app/src/components/Page.js [hunk 4]
```diff
@@ -1,14 +1,26 @@
...
+      </div>
+      {modals}
```
Comment (mcous) [inline app/src/components/Page.js]: @IanLondon and @b-cooper I wanted to pick y'all brains about thie `modals` prop. Requirements:

1. A `<Page>` has children, and if those children overflow they should be scrollable
2. A `<Page>` may also display modals, and the modal should always absolutely fill the `<Page>` and it should never scroll
    - Aside: the modal children may scroll; I'm talking about the weird current behavior where you can scroll the overlay away if the window is tool short to fit the page

Basically, we have two sets of children for a page, 1 set that should be wrapped in a scrollable div and one set that should live outside that div. Do we think a `React.Node` prop is the best way to handle this?

File: app/src/components/Page.css [hunk 5]
```diff
@@ -1,5 +1,9 @@
...
   position: relative;
-  overflow-y: scroll;
+}
+
+.main {
```
Comment (mcous) [inline app/src/components/Page.css]: The `main` class should have `overflow-y: auto`

File: app/src/components/RobotSettings/styles.css [hunk 6]
```diff
@@ -9,6 +9,7 @@
   overflow-y: auto;
```
Comment (mcous) [inline app/src/components/RobotSettings/styles.css]: `overflow-y: auto;` should be removed here in favor of the overflow rule on `.main`

File: app/src/components/RobotSettings/styles.css [hunk 7]
```diff
@@ -9,6 +9,7 @@
...
   padding: calc(2 * var(--robot_settings_card_spacing));
   padding-bottom: 0;
+  max-height: 100%;
```
Comment (mcous) [inline app/src/components/RobotSettings/styles.css]: This rule is unnecessary because `.robot_settings` should not be scrollable, it's parent (`.main`) should

File: components/src/modals/modals.css [hunk 8]
```diff
@@ -5,10 +5,12 @@
   --modal-contents: {
     z-index: 1;
     width: 100%;
+    max-height: 100%;
     max-width: 38rem;
-    margin: auto;
+    margin: 0 auto;
     padding: 1rem;
     border-radius: 0.5rem;
+    overflow-y: scroll;
```
Comment (mcous) [inline components/src/modals/modals.css]: Should this be `auto`?

File: components/src/modals/modals.css [hunk 9]
```diff
@@ -43,6 +56,7 @@
 
   background-color: white;
   min-width: 38rem;
+  overflow-y: scroll;
```
Comment (mcous) [inline components/src/modals/modals.css]: Redundant; already applied from `--modal-contents`

File: app/src/components/Page.js (hunk 2 above)
Comment (Kadee80) [reply app/src/components/Page.js]: Had it as react.node - wound up breaking on calibrate instruments page. will follow up on slack about this one

File: app/src/components/Page.css (hunk 1 above)
Comment (Kadee80) [reply app/src/components/Page.css]: turns out the flex basis isn't needed at all.

File: app/src/components/Page.css (hunk 5 above)
Comment (Kadee80) [reply app/src/components/Page.css]: 👍

File: app/src/components/Page.js [hunk 10]
```diff
@@ -1,14 +1,28 @@
...
   return (
     <main className={styles.task}>
-      {props.children}
+      {titleBar && (
+        <TitleBar {...titleBar} />
+      )}
+      <div className={styles.main}>
+        {children}
+      </div>
+      {modals}
```
Comment (Kadee80) [inline app/src/components/Page.js]: @IanLondon and @b-cooperreposing mikes question about the `modals` prop. 

Requirements:
1. A `<Page>` has children, and if those children overflow they should be scrollable
2. A `<Page>` may also display modals, and the modal should always absolutely fill the `<Page>` and it should never scroll
    • Aside: the modal children may scroll; I'm talking about the weird current behavior where you can scroll the overlay away if the window is tool short to fit the page

Basically, we have two sets of children for a page, 1 set that should be wrapped in a scrollable div and one set that should live outside that div. Do we think a `React.Node` prop is the best way to handle this?

**Comment by IanLondon** [comment]
I don't remember seeing this kind of scrollbar before, is it OK?

image

**Comment by Kadee80** [comment]
Closing this PR in favor of a different solution the team came up with: Modals as siblings of Page, not children.
